### PR TITLE
feature: Add environment variable support for SageMaker training job

### PR DESCRIPTION
**Pull request by chiaengc**
Do not merge until the corresponding API changes are released

*Issue #, if available:*
https://github.com/aws/sagemaker-python-sdk/issues/2207

*Description of changes:* Add environment variable support for SageMaker training job

*Testing done:*
test: tox tests/unit

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [X] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [X] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [X] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [X] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jswudi — COMMENTED** [review]
Does the backend work ready to support env var? If yes, do we need to add env in one integration test?

File: tests/unit/test_estimator.py [hunk 1]
```diff
@@ -2677,6 +2682,39 @@ def test_generic_to_fit_with_sagemaker_metrics_missing(sagemaker_session):
...
+        output_path=OUTPUT_PATH,
+        sagemaker_session=sagemaker_session,
+        environment=ENV_INPUT,
+    )
+
+    e.fit()
+
+    sagemaker_session.train.assert_called_once()
+    args = sagemaker_session.train.call_args[1]
+    assert args["environment"]
```
[the remark this answers is not in the document]
Comment (chiaengc) [reply tests/unit/test_estimator.py]: Done.

File: tests/unit/test_estimator.py [hunk 2]
```diff
@@ -2677,6 +2682,39 @@ def test_generic_to_fit_with_sagemaker_metrics_missing(sagemaker_session):
...
+
+def test_no_environment_variables_in_train_args(sagemaker_session):
+    e = Estimator(
+        IMAGE_URI,
+        ROLE,
+        INSTANCE_COUNT,
+        INSTANCE_TYPE,
+        output_path=OUTPUT_PATH,
+        sagemaker_session=sagemaker_session,
+        environment=None,
```
[the remark this answers is not in the document]
Comment (chiaengc) [reply tests/unit/test_estimator.py]: Removed this test

File: src/sagemaker/estimator.py [hunk 3]
```diff
@@ -1834,6 +1841,7 @@ def __init__(
             enable_network_isolation=enable_network_isolation,
             profiler_config=profiler_config,
             disable_profiler=disable_profiler,
+            environment=environment,
```
[the remark this answers is not in the document]
Comment (chiaengc) [reply src/sagemaker/estimator.py:1846]: Added doc for environment

**Review by icywang86rui — COMMENTED** [review]
Please add an integ test too. You can add on to an existing test.

File: src/sagemaker/estimator.py [hunk 4]
```diff
@@ -263,6 +264,8 @@ def __init__(
                 ``disable_profiler`` parameter to ``True``.
             disable_profiler (bool): Specifies whether Debugger monitoring and profiling
                 will be disabled (default: ``False``).
+            environment (dict[str, str]) : A string to string map contains environment
+                variables to set in the Docker container.
```
[the remark this answers is not in the document]
Comment (chiaengc) [reply src/sagemaker/estimator.py]: Done

File: src/sagemaker/estimator.py [hunk 5]
```diff
@@ -1801,6 +1808,8 @@ def __init__(
...
```
[the remark this answers is not in the document]
Comment (chiaengc) [reply src/sagemaker/estimator.py]: Done

File: src/sagemaker/session.py [hunk 6]
```diff
@@ -525,6 +526,8 @@ def train(  # noqa: C901
             profiler_rule_configs (list[dict]): A list of profiler rule configurations.
             profiler_config (dict): Configuration for how profiling information is emitted
                 with SageMaker Profiler. (default: ``None``).
+            environment (dict[str, str]) : A string to string map contains environment
+                variables to set in the Docker container.
```
[the remark this answers is not in the document]
Comment (chiaengc) [reply src/sagemaker/session.py]: Done

File: src/sagemaker/session.py [hunk 7]
```diff
@@ -657,6 +662,8 @@ def _get_train_request(  # noqa: C901
             profiler_rule_configs (list[dict]): A list of profiler rule configurations.
             profiler_config(dict): Configuration for how profiling information is emitted with
                 SageMaker Profiler. (default: ``None``).
+            environment (dict[str, str]) : A string to string map contains environment
+                variables to set in the Docker container.
```
[the remark this answers is not in the document]
Comment (chiaengc) [reply src/sagemaker/session.py]: Done

**Review by icywang86rui — APPROVED** [review]

**Pull request merged**